outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -89,8 +89,9 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
             {
                 const enrichmentInfo = new EnrichmentInfo(parsed.result['enrichment_info'].type,
                     parsed.result['enrichment_info'].info);
-                if (enrichmentInfo.type != EnrichmentInfoType.NoData)
-                    this.context.showTopBarMessage(enrichmentInfo);
+                if (enrichmentInfo.type == EnrichmentInfoType.NoData)
+                    this.enriched = true;
```

Comment:
Remember to always use brackets even for one-line condition as a best practice

```js
if (enrichmentInfo.type == EnrichmentInfoType.NoData) {
    this.enriched = true;
}
```

Reply:
I find it a little strange that the "enriched" boolean is placed on a "CompanySection" and not on the Company object itself.

If it's used to also store the fact that we "attempted" to enrich the company and we receive a "EnrichmentInfoType.NoData", then maybe it should be named otherwise, like "isEnrichmentRequestEmpty" ? So we can actually differentiate those 2 states.

I would actually prefer if you had a "isEnriched" method/property on your Company object instead of your current "hasInsights", the "enrich" word is more speaking in terms of IAP etc.

(And then nothing prevents you from having a "showEnrichButton" method that returns False if "isEnrichmentRequestEmpty" or "company.isEnriched()")

What do you think about it?
Purpose is to have meaningful variables and a clear structure / state.

EDIT:

And now I discover that the "isEmpty()" method on the company means that it could not be enriched, which is even more confusin.
We could instead have a "enrichmentStatus" property / method that behaves like a selection field:
"enrichment_available" -> when you have not enriched it yet
"enriched" -> when it's enriched and filled up with data
"enrichment_empty" -> when the enrichment request returns no data.

Just some wild thoughts, I don't know the whole structure but I think we can improve the readability a bit.

Reply:
Yes indeed, this had to be changed for more readability.

We now use an enrichmentStatus property which enables us to more easily handle company enrichment states.

Can you please have a look again at this and tell me what do you think about?